test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/IrisPlantClassificationTests.cs

```diff
@@ -22,56 +22,43 @@ public partial class ScenariosTests
         [Fact]
         public void TrainAndPredictIrisModelUsingDirectInstantiationTest()
         {
-            string dataPath = GetDataPath("iris.txt");
-            string testDataPath = dataPath;
+            var mlContext = new MLContext(seed: 1, conc: 1);
 
-            using (var env = new ConsoleEnvironment(seed: 1, conc: 1))
+            var reader = mlContext.Data.TextReader(new TextLoader.Arguments()
             {
-                // Pipeline
-                var loader = TextLoader.ReadFile(env,
-                    new TextLoader.Arguments()
-                    {
-                        HasHeader = false,
-                        Column = new[]
-                        {
-                            new TextLoader.Column("Label", DataKind.R4, 0),
-                            new TextLoader.Column("SepalLength", DataKind.R4, 1),
-                            new TextLoader.Column("SepalWidth", DataKind.R4, 2),
-                            new TextLoader.Column("PetalLength", DataKind.R4, 3),
-                            new TextLoader.Column("PetalWidth", DataKind.R4, 4)
-                        }
-                    }, new MultiFileSource(dataPath));
-
-                IDataView pipeline = new ColumnConcatenatingTransformer(env, "Features",
-                    "SepalLength", "SepalWidth", "PetalLength", "PetalWidth").Transform(loader);
-
-                // NormalizingEstimator is not automatically added though the trainer has 'NormalizeFeatures' On/Auto
-                pipeline = NormalizeTransform.CreateMinMaxNormalizer(env, pipeline, "Features");
-
-                // Train
-                var trainer = new SdcaMultiClassTrainer(env, "Label", "Features", advancedSettings: (s) => s.NumThreads = 1);
-
-                // Explicity adding CacheDataView since caching is not working though trainer has 'Caching' On/Auto
-                var cached = new CacheDataView(env, pipeline, prefetch: null);
-                var trainRoles = new RoleMappedData(cached, label: "Label", feature: "Features");
-                var pred = trainer.Train(trainRoles);
-
-                // Get scorer and evaluate the predictions from test data
-                IDataScorerTransform testDataScorer = GetScorer(env, pipeline, pred, testDataPath);
-                var metrics = Evaluate(env, testDataScorer);
-                CompareMatrics(metrics);
-
-                // Create prediction engine and test predictions
-                var model = env.CreatePredictionEngine<IrisData, IrisPrediction>(testDataScorer);
```

Comment:
>CreatePredictionEngine [](start = 32, length = 22)

don't you want to hide it?

Reply:
What do you mean? This PR aims at removing CreatePredictionEigine so it should be removed instead of being hidden.

---
In reply to: [234297716](https://github.com/dotnet/machinelearning/pull/1637#discussion_r234297716) [](ancestors = 234297716)

Reply:
Ok, do you want to remove it? It's just strange what you remove usage and leave class public

---
In reply to: [234300118](https://github.com/dotnet/machinelearning/pull/1637#discussion_r234300118) [](ancestors = 234300118,234297716)

Reply:
Those functions are removed now. Thanks.

---
In reply to: [234302075](https://github.com/dotnet/machinelearning/pull/1637#discussion_r234302075) [](ancestors = 234302075,234300118,234297716)